Habitat profile bug fixes and improvements

rhass opened the pull request:
* Add `core/hab` to `pkg_deps` in hab plan.
  This fixes a bug in which tests needing to execute things like `hab pkg path myorigin/mypath` in the can profile/test can successfully execute the command. This was caused by the runtime of `hab pkg exec chef/inspec` which changes the path for the inspec runtime to match that of the inspec hab package.

* Fixed bug with install step where profile would include the .hart files from previous builds.
* Added ruby to pkg_deps rather than adding it to the path via the runhook. This ensures `hab pkg exec chef/inspec` behaves in the same way as the run hook.
* Updated the generated plan to support plan.sh syntax changes in habitat 0.21.0 and later by removing the `pkg_source` and the `do_download`, `do_verify`, and `do_unpack` overrides.
* Updated the generate run hook to leverage habitat to perform most of the origin, package name, and path variable interpolations.

/cc @schisamo @elliott-davis

rhass committed:
Add `core/hab` to `pkg_deps` in hab plan.

Runtime of `hab pkg exec chef/inspec` changes the path for the inspec
runtime to match that of the inspec hab package. This makes it so tests
which need to execute things like `hab pkg path myorigin/mypath` in the
can profile/test can successfully execute the command.

Signed-off-by: Ryan Hass <rhass@users.noreply.github.com>

rhass committed:
Update habitat plan and hook generators

* Fixed bug with install step where profile would include the .hart
  files from previous builds.
* Updated the generated plan to support plan.sh syntax changes in
  habitat 0.21.0 and later by removing the `pkg_source` and the
  `do_download`, `do_verify`, and `do_unpack` overrides.
* Updated the generate run hook to leverage habitat to perform most of
  the origin, package name, and path variable interpolations.

Signed-off-by: Ryan Hass <rhass@users.noreply.github.com>

adamleff commented on lib/bundles/inspec-habitat/profile.rb:
```diff
@@ -336,23 +332,21 @@ def run_hook_contents
 export HOME={{pkg.svc_var_path}}
 
-PROFILE_IDENT="#{habitat_origin}/#{package_name}"
+PROFILE_IDENT="{{pkg.origin}/{{pkg.name}}"
```
Missing a `}` for `{{pkg.origin}}`... without it, the sup fails with:

```
Failed to load hook: hab-sup(ER)[src/error.rs:355:8]: TemplateError(TemplateError { reason: InvalidSyntax, template_name: Some("hook"), line_no: Some(7), column_no: Some(28) })
```

... and then the sup fails because the run hook never got copied to `/hab/svc`

```
Service start failed: hab-sup(ER)[src/error.rs:379:8]: No such file or directory (os error 2)
```

adamleff reviewed (changes requested):
@rhass Love it. Need to fix one breakage though and then we should be good.

rhass committed:
Fix exit status check in hab run hook generator.

The exit status would never return "InSpec run completed successfully"
since the value of $RC was always an integer which never was prefixed
with an "x". This checks the return directly since we currently do not
have any complex logic which warrants the need to check different
return status values where a prefixed return code is necessary.

Signed-off-by: Ryan Hass <rhass@users.noreply.github.com>

chris-rock reviewed (approved):
Thank you @rhass
